helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -859,13 +866,18 @@ public boolean isGlobalRebalanceAsyncModeEnabled() {
 
   /**
    * Set the abnormal state resolver class map.
+   * @param resolverMap - the resolver map
+   *                    If null, the resolver map item will be removed from the config.
    */
   public void setAbnormalStateResolverMap(Map<String, String> resolverMap) {
-    if (resolverMap.values().stream()
-        .anyMatch(className -> className == null || className.isEmpty())) {
+    if (resolverMap != null && resolverMap.values().stream()
```

Comment:
I am a little bit confused. 

`anyMatch(className -> className == null))` means if there is a className (key) in resolverMap, we throw exception. 

Then how about value in resolverMap, can it be null? What does it mean then?

Reply:
It is searching through the value set. The stream() is not done to the key set. I think you missed the first part.

But it is a good catch that I should not remove "className.isEmpty()" check.

Reply:
Right, we are saying value can't be empty or 'null'. However, don't forget about key, the key in resovlerMap is potentially can be null too, right? Shall we also throw an exception there?